omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,292 @@
+		description.addRequiredProperty("objectUuid");
+		
+		return description;
+	}
+	
+	/**
+	 * @see DelegatingCrudResource#doGetAll(RequestContext)
+	 */
+	@Override
+	protected NeedsPaging<Tag> doGetAll(RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	/**
+	 * Method to obtain the Uri of any OpenrmrsObject
+	 * 
+	 * @param objectType the Java class name of the object
+	 * @param objectUuid the Uuid of the object
+	 * @return the Uri of the given object
+	 */
+	private String getUriOfObject(String objectType, String objectUuid) {
+		Class<?> className = null;
+		try {
+			className = Context.loadClass(objectType);
+		}
+		catch (ClassNotFoundException e) {}
+		BaseDelegatingResource resource = (BaseDelegatingResource) Context.getService(RestService.class)
+		        .getResourceBySupportedClass(className);
+		return resource.getUri(resource.getByUniqueId(objectUuid));
+	}
+	
+	/**
+	 * Extracts resource name from a resource handler
+	 * 
+	 * @return resource name
+	 */
+	private String getResourceName(String objectType) {
+		
+		Class<?> className = null;
+		try {
+			className = Context.loadClass(objectType);
+		}
+		catch (ClassNotFoundException e) {}
+		DelegatingResourceHandler resourceHandler = (DelegatingResourceHandler<?>) Context.getService(RestService.class)
+		        .getResourceBySupportedClass(className);
+		Resource annotation = resourceHandler.getClass().getAnnotation(Resource.class);
+		
+		String resourceName = null;
+		if (annotation != null) {
+			resourceName = annotation.name();
+		} else {
+			SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
+			if (subResourceAnnotation != null) {
+				resourceName = subResourceAnnotation.path();
+			}
+		}
+		return resourceName;
+	}
+	
+	/**
+	 * @see DelegatingCrudResource#doSearch(RequestContext)
+	 */
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		String tagName = context.getRequest().getParameter("tag");
+		String objectType = context.getRequest().getParameter("objectType");
+		String objectUuid = context.getRequest().getParameter("objectUuid");
+		if (StringUtils.isNotBlank(objectType)) {
+			ArrayList<Class<? extends OpenmrsObject>> openmrsObjects = new ArrayList<Class<? extends OpenmrsObject>>();
+			try {
+				openmrsObjects.add((Class<? extends OpenmrsObject>) OpenmrsClassLoader.getInstance().loadClass(objectType));
+			}
+			catch (ClassNotFoundException e) {
+				log.error("Class of" + objectType + "Not Found", e);
+			}
+			if (StringUtils.isNotBlank(tagName)) {
+				ArrayList<String> tagNames = new ArrayList<String>();
+				tagNames.add(tagName);
+				return new NeedsPaging<Tag>(getService().getTags(openmrsObjects, tagNames), context);
+			} else if (StringUtils.isNotBlank(objectUuid)) {
+				return new NeedsPaging<Tag>(getService().getTags(objectType, objectUuid), context);
+			} else {
+				throw new ObjectNotFoundException();
```

Comment:
You are throwing the wrong exception because it will be deceptive, this is a bad request so you need to throw an exception along these lines, I think `IllegalRequestException` is better